Branch-SDK/src/io/branch/referral/RemoteInterface.java

```diff
@@ -26,128 +27,183 @@
-				if (ex instanceof NetworkOnMainThreadException)
-					Log.i("BranchSDK", "Branch Error: Don't call our synchronous methods on the main thread!!!");
-			}
-			return new ServerResponse(tag, 500);
-		}
-	}
+    public static final String NO_TAG_VALUE = "no_tag";
+    public static final int NO_CONNECTIVITY_STATUS = -1009;
+    public static final int NO_API_KEY_STATUS = -1234;
+
+    private static final String SDK_VERSION = "1.4.4";
+    private static final int DEFAULT_TIMEOUT = 3000;
+
+    protected PrefHelper prefHelper_;
+
+    public RemoteInterface() { }
+
+    public RemoteInterface(Context context) {
+        prefHelper_ = PrefHelper.getInstance(context);
+    }
+
+    private HttpClient getGenericHttpClient(int timeout) {
+        if (timeout <= 0) {
+            timeout = DEFAULT_TIMEOUT;
+        }
+        HttpParams httpParams = new BasicHttpParams();
+        HttpConnectionParams.setConnectionTimeout(httpParams, timeout);
+        HttpConnectionParams.setSoTimeout(httpParams, timeout);
+        return new DefaultHttpClient(httpParams);
+    }
+
+    private ServerResponse processEntityForJSON (HttpEntity entity, int statusCode, String tag, boolean log, BranchLinkData linkData) {
+        ServerResponse result = new ServerResponse(tag, statusCode, linkData);
+        try {
+            if (entity != null) {
+                InputStream instream = entity.getContent();
+                BufferedReader rd = new BufferedReader(new InputStreamReader(instream));
+
+                String line = rd.readLine();
+                if (log) {
+                    PrefHelper.Debug(BranchConfig.TAG, "returned " + line);
+                }
+
+                if (line != null) {
+                    try {
+                        JSONObject jsonObj = new JSONObject(line);
+                        result.setPost(jsonObj);
+                    } catch (JSONException ex) {
+                        try {
+                            JSONArray jsonArray = new JSONArray(line);
+                            result.setPost(jsonArray);
+                        } catch (JSONException ex2) {
+                            if (log) {
+                                PrefHelper.Debug(getClass().getSimpleName(), "JSON exception: "
+                                    + ex2.getMessage());
+                            }
+                        }
+                    }
+                }
+            }
+        } catch (IOException ex) {
+            if (log) {
+                PrefHelper.Debug(getClass().getSimpleName(), "IO exception: "
+                        + ex.getMessage());
+            }
+        }
+        return result;
+    }
+
+    //region GET Requests
+    public ServerResponse make_restful_get(String url, String tag, int timeout) {
+        return make_restful_get(url, tag, timeout, 0, true);
+    }
+
+    public ServerResponse make_restful_get(String url, String tag, int timeout, int retryNumber, boolean log) {
```

Comment:
i thought about this too. the base method should be private for both GETs and POSTs. I'll make this change as well.